testing/tests/DevExpress.ui.widgets.dataGrid/dataGrid.tests.js

```diff
@@ -178,6 +178,33 @@ QUnit.test("Accessibility columns id should not set for columns editors (T710132
     assert.equal($(".dx-texteditor [id]").length, 0, "editors has no accessibility id");
 });
 
+QUnit.test("DataGrid - Should hide filter row menu after losing it's focus", function(assert) {
+    // arrange
+    var filterRowWrapper = new FilterRowWrapper(".dx-datagrid"),
+        $menu,
+        $root,
+        menuInstance;
+
+    createDataGrid({
+        filterRow: { visible: true },
+        dataSource: [{ field1: "1", field2: "2" }]
+    });
+    this.clock.tick();
+
+    // act
+    $menu = filterRowWrapper.getMenuElement(0);
+    $menu.focus();
+
+    menuInstance = $menu.dxMenu("instance");
+    $root = $(menuInstance.itemElements().get(0));
+    menuInstance._showSubmenu($root);
+
+    $menu.trigger("blur");
+
+    // assert
+    assert.notOk(!!menuInstance._visibleSubmenu, "submenu is hidden");
```

Comment:
It is bad assert. If _visibleSubmenu is renamed test will not fail. It is better to check menu overlay element visiibility

Reply:
I added double check for the submenu visibility which covers the "private fields was renamed" case.